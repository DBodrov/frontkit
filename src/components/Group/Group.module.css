.group {
    display: flex;
    width: 100%;
}

.group > * {
    border-radius: 0 !important;
}

.group > * + * {
    border-left: 1px solid #d5dae0;
}

.group > :last-child {
    border-radius: 0 8px 8px 0 !important;
}

.group > :first-child {
    border-radius: 8px 0 0 8px !important;
}
