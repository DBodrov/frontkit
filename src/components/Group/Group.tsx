import React from 'react';

import styles from './Group.module.css';

export function Group(props: React.BaseHTMLAttributes<HTMLSpanElement>): JSX.Element {
    const { children } = props;
    return <div className={styles.group}>{children}</div>;
}
