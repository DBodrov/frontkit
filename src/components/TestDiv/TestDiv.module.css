.block {
    height: 40px;
    border-radius: 8px;
    width: 100%;
    background-color: #eaeef4;
}
