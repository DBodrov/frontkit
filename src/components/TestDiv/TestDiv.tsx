import React from 'react';

import styles from './TestDiv.module.css';

export function TestDiv(props: React.BaseHTMLAttributes<HTMLSpanElement>): JSX.Element {
    const { children } = props;
    return <div className={styles.block}>{children}</div>;
}
