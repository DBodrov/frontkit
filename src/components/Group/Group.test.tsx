import React from 'react';
import ReactDOM from 'react-dom';
import { TestDiv } from '../TestDiv/TestDiv';
import { Group } from './Group';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Group>
            <TestDiv>
                <TestDiv />
                <TestDiv />
                <TestDiv />
            </TestDiv>
            <TestDiv />
            <TestDiv />
        </Group>,
        div,
    );
    expect(div).toMatchSnapshot();
});
