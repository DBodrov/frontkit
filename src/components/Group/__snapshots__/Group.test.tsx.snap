// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="group"
    data-test-id="Group-Container"
  >
    <div
      class="block"
    >
      <div
        class="block"
      />
      <div
        class="block"
      />
      <div
        class="block"
      />
    </div>
    <div
      class="block"
    />
    <div
      class="block"
    />
  </div>
</div>
`;
